server/events/models/models.go

```diff
@@ -488,3 +488,25 @@ type WorkflowHookCommandContext struct {
 	// Verbose is true when the user would like verbose output.
 	Verbose bool
 }
+
+type EnqueueStatusType int
+
+const (
+	// Enqueued means the ProjectLock entered the queue
+	Enqueued EnqueueStatusType = iota
+
+	// AlreadyInTheQueue means the ProjectLock is already in the queue
+	AlreadyInTheQueue
+)
+
+type EnqueueStatus struct {
+	// Status is the status of the enqueue operation
+	Status EnqueueStatusType
+	// ProjectLocksInFront tells how many PRs are in line before the current one
+	ProjectLocksInFront int
```

Comment:
Just nit. Maybe we can rename `ProjectLocksInFront` to `QueueDepth`? Just by looking at `EnqueueStatus.ProjectLocksInFront`, it might seem like the real lock objects?

Reply:
Thanks, renamed https://github.com/runatlantis/atlantis/pull/2782/commits/06a83e3570eae401f2913dd8b7893919e271e1c2 (https://github.com/runatlantis/atlantis/pull/2782)